depclean-gradle-plugin/src/main/java/se/kth/depclean/DepCleanGradleAction.java

```diff
@@ -0,0 +1,852 @@
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import org.apache.commons.io.FileUtils;
+import org.gradle.api.Action;
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.ConfigurationContainer;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.gradle.api.artifacts.ResolvedDependency;
+import org.gradle.api.artifacts.UnresolvedDependency;
+import org.gradle.api.logging.Logger;
+import org.jetbrains.annotations.NotNull;
+import se.kth.depclean.analysis.DefaultGradleProjectDependencyAnalyzer;
+import se.kth.depclean.analysis.GradleProjectDependencyAnalysis;
+import se.kth.depclean.core.analysis.ProjectDependencyAnalyzerException;
+import se.kth.depclean.util.JarUtils;
+
+/**
+ * Depclean default and only action.
+ */
+public class DepCleanGradleAction implements Action<Project> {
+
+  // To get some clear visible results.
+  private static final String SEPARATOR = "-------------------------------------------------------";
+
+  // To reduce some code(Frequent use)
+  private static final String Sep = File.separator;
+
+  /**
+   * A map [artifact] -> [configuration].
+   */
+  private static final Map<ResolvedArtifact,
+          String> ArtifactConfigurationMap = new HashMap<>();
+
+  /**
+   * A map [dependencies] -> [size].
+   */
+  private static final Map<String, Long> SizeOfDependencies = new HashMap<>();
+
+  // Extensions fields =====================================
+  private Project project;
+  private boolean skipDepClean;
+  private boolean createBuildDebloated;
+  // TODO : The implementation of next two parameters will be done later.
+  private boolean createResultJson;
+  private boolean createClassUsageCsv;
+  private boolean isIgnoreTest;
+  private boolean failIfUnusedDirect;
+  private boolean failIfUnusedTransitive;
+  private boolean failIfUnusedInherited;
+  private Set<String> ignoreConfiguration;
+  private Set<String> ignoreDependencies;
+
+  @SneakyThrows
+  @Override
+  public void execute(@NotNull Project project) {
+
+    Logger logger = project.getLogger();
+
+    // If the user provided some configuration.
+    DepCleanGradlePluginExtension extension = project.getExtensions()
+            .getByType(DepCleanGradlePluginExtension.class);
+    getPluginExtensions(extension);
+
+    if (skipDepClean) {
+      logger.lifecycle("Skipping DepClean plugin execution");
+      return;
+    }
+
+    // If the project is not the default one.
+    if (this.project != null) {
+      project = this.project;
+    }
+
+    // Path to the project directory.
+    final String projectDirPath = project.getProjectDir().getAbsolutePath() + Sep;
+
+    // Path to the dependency directory.
+    final String dependencyDirPath = projectDirPath + "build" + Sep + "Dependency";
```

Comment:
You could use Paths.get: https://docs.oracle.com/javase/7/docs/api/java/nio/file/Paths.html#get(java.lang.String,%20java.lang.String...)

Reply:
Thanks. that's a nice refactoring. I will implement it (wherever possible) in my next push.